tests/test_api_groups_update.py

```diff
@@ -177,24 +192,27 @@ def test_patch_group_hosts_from_different_group(
 
 @pytest.mark.usefixtures("enable_rbac", "event_producer")
 def test_patch_groups_RBAC_allowed_specific_groups(mocker, db_create_group_with_hosts, api_patch_group):
-    get_rbac_permissions_mock = mocker.patch("lib.middleware.get_rbac_permissions")
-    group_id = str(db_create_group_with_hosts("new_group", 3).id)
-    # Make a list of allowed group IDs (including some mock ones)
-    group_id_list = [generate_uuid(), group_id, generate_uuid()]
-
-    # Grant permissions to all 3 groups
-    mock_rbac_response = create_mock_rbac_response(
-        "tests/helpers/rbac-mock-data/inv-groups-write-resource-defs-template.json"
-    )
-    mock_rbac_response[0]["resourceDefinitions"][0]["attributeFilter"]["value"] = group_id_list
-
-    get_rbac_permissions_mock.return_value = mock_rbac_response
-    patch_doc = {"name": "new_name"}
-
-    response_status, _ = api_patch_group(group_id, patch_doc)
-
-    # Should be allowed
-    assert_response_status(response_status, 200)
+    with TemporaryDirectory() as temp_directory:
+        with set_environment({"PROMETHEUS_MULTIPROC_DIR": temp_directory}):
+            get_rbac_permissions_mock = mocker.patch("lib.middleware.get_rbac_permissions")
+            patch_rbac_workspace_mock = mocker.patch("api.group.patch_rbac_workspace")  # noqa: F841, this variable used by the test framework
```

Comment:
Avoid using noqa when you can. In this case, don't you just need to remove the variable?
```suggestion
            mocker.patch("api.group.patch_rbac_workspace")
```